### PR TITLE
[DOCS] Reorganizes Transforms limitations

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR arranges the transform-related limitations into three groups (and adds a short explanation about them to the intro):
    * Configuration
    * Operation
    * UI

This PR is part of an effort that aims to make the AD, DFA, and Transforms limitations sections similar to each other in structure. 
Related PRs:
* https://github.com/elastic/stack-docs/pull/1471
* https://github.com/elastic/stack-docs/pull/1607

### Preview

[Transform limitations](https://elasticsearch_70638.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/transform-limitations.html)

### CODE REVIEW

File: docs/reference/transform/limitations.asciidoc [hunk 1]
```diff
@@ -248,40 +277,32 @@ documents if your {transform} contains a `group_by` based on `date_histogram`.
...
+If your cluster contains mixed version nodes, for example during a rolling 
+upgrade from 7.2 to a newer version, and {transforms} have been created in 7.2, 
+the {transforms} UI (earler {dataframe} UI) will not work. Please wait until all 
+nodes have been upgraded to the newer version before using the {transforms} UI.
 
 [discrete]
-[[transforms-ccs-limitation]]
-=== {transforms-cap} support {ccs} if the remote cluster is configured properly
+[[transform-kibana-limitations]]
+=== Up to 1,000 {transforms} are supported
```
Comment (hendrikmuhs) [inline docs/reference/transform/limitations.asciidoc:303]: I am not sure about this limitation, at least in the backend this limitation got removed some time ago. Would be good to verify: @elastic/ml-ui is this still a limitation?

Comment (peteharverson) [reply docs/reference/transform/limitations.asciidoc:303]: The Transforms list in the UI still passes a `size` parameter of `1000` to the `GET` transforms endpoint, so suggest changing that text to indicate it's only a limit to the number of transforms that are listed in the Transform management page in the UI.

**Comment by szabosteve** [comment]
@hendrikmuhs I removed the datatype-related outdated item and amended the one about the number of transforms listed in Kibana via  71e0294.

**Review by hendrikmuhs — APPROVED** [review]
Thanks for having another look!

LGTM

File: docs/reference/transform/limitations.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/transform/limitations.asciidoc]: Thanks for the help, @peteharverson!

**Pull request merged**